### PR TITLE
docs: add GetImplicitResourcesForUser and GetImplicitUsersForRole

### PR DESCRIPTION
**Pull request by closetool**
Signed-off-by: closetool <4closetool3@gmail.com>

Fix: casbin/casbin#798

### CODE REVIEW

**Comment by hsluoyz** [comment]
@ErikQQY plz review.

**Review by ErikQQY — APPROVED** [review]
LGTM

File: docs/RBACAPI.md [hunk 1]
```diff
@@ -638,6 +638,30 @@ e.get_implicit_roles_for_user("alice", None); // No domain
...
+But GetImplicitRolesForUser("alice") will get: ["role:admin", "alice"].
+
+
+For example:
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--Go-->
+```go
+users := e.GetImplicitRolesForUser("role:user")
```
Comment (hsluoyz) [inline docs/RBACAPI.md:660]: Is it on earth `GetImplicitUsersForRole()` or `GetImplicitRolesForUser()`?

File: docs/RBACAPI.md [hunk 2]
```diff
@@ -638,6 +638,30 @@ e.get_implicit_roles_for_user("alice", None); // No domain
...
+### `GetImplicitUsersForRole()`
+
+GetImplicitUsersForRole gets all users inheriting the role.
+Compared to GetUsersForRole(), this function retrieves indirect users.
+
+For example:  
+g, alice, role:admin  
+g, role:admin, role:user  
+
+Get("role:user") can only get: ["role:admin"].
```
Comment (hsluoyz) [inline docs/RBACAPI.md:650]: What's `Get()`?

File: docs/RBACAPI.md [hunk 3]
```diff
@@ -638,6 +638,30 @@ e.get_implicit_roles_for_user("alice", None); // No domain
...
+
+GetImplicitUsersForRole gets all users inheriting the role.
+Compared to GetUsersForRole(), this function retrieves indirect users.
+
+For example:  
+g, alice, role:admin  
+g, role:admin, role:user  
+
+Get("role:user") can only get: ["role:admin"].  
+But GetImplicitRolesForUser("alice") will get: ["role:admin", "alice"].
```
Comment (hsluoyz) [inline docs/RBACAPI.md:651]: Are you sure `GetImplicitRolesForUser("alice") will get: ["role:admin", "alice"].` is correct?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by closetool** [comment]
@hsluoyz fixed, sorry for my mistakes.

File: docs/RBACAPI.md [hunk 4]
```diff
@@ -638,6 +638,30 @@ e.get_implicit_roles_for_user("alice", None); // No domain
...
+
+GetImplicitUsersForRole gets all users inheriting the role.
+Compared to GetUsersForRole(), this function retrieves indirect users.
+
+For example:  
+g, alice, role:admin  
+g, role:admin, role:user  
+
+GetUsersForRole("role:user") can only get: ["role:admin"].  
+But GetImplicitUesrsForRole("alice") will get: ["role:admin", "alice"].
```
Comment (hsluoyz) [inline docs/RBACAPI.md:651]: Wrong here?

image

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**